response/page_path.go

```diff
@@ -2,13 +2,127 @@ package response
 
 import (
 	"context"
+	"net/url"
+	"regexp"
+	"strings"
 
 	"github.com/ONSdigital/log.go/v2/log"
 )
 
+var visualisationsEndpointRegexp = regexp.MustCompile(`(^/visualisations/.+?)/`)
+var bulletinsOrArticlesPathRegexp = regexp.MustCompile(`(.+/(bulletin|article)s/.+?/[^/]+)`)
+var methodologiesOrQMIsOrAdhocsPathRegexp = regexp.MustCompile(`.+/(methodologies|qmis|adhocs)/([^/]+)`)
+var endsWithFileExtRegexp = regexp.MustCompile(`/[^/]+\.\w+$`)
+var timeSeriesPathRegexp = regexp.MustCompile(`^(.*?/timeseries(?:/[^/]+){0,2})`)
+var datasetsPathRegexp = regexp.MustCompile(`^(.*?/datasets(?:/[^/]+){0,2})`)
+
 func getPagePath(ctx context.Context, uri string) string {
 	log.Info(ctx, "Calculating page path for "+uri)
```

Comment:
It's good that you're thinking about edge cases :-) But `uri` cannot be `nil`, it's always a string. Trimming is not necessary either, because this comes directly from the request, so it will be trimmed already.